newsroom/agenda/service.py

```diff
@@ -0,0 +1,120 @@
+        display_from and display_to.
+        """
+        for item in docs:
+            date = datetime.strptime(item.id, "%Y%m%d")
+            item.display_from = local_to_utc(item.tz, date.replace(hour=0, minute=0, second=0))
+            item.display_to = local_to_utc(item.tz, date.replace(hour=23, minute=59, second=59))
+        await super().on_create(docs)
+
+    async def find_one_for_date(self, for_date: datetime) -> FeaturedResourceModel | None:
+        return await self.find_one(display_from={"$lte": for_date}, display_to={"$gte": for_date})
+
+    async def get_featured_stories(
+        self, date_from: str, timezone_offset: int = 0, query_string: Optional[str] = None, from_offset: int = 0
+    ) -> Any:
+        for_date = datetime.strptime(date_from, "%d/%m/%Y %H:%M")
+        local_date = get_local_date(
+            for_date.strftime("%Y-%m-%d"),
+            for_date.strftime("%H:%M:%S"),
+            timezone_offset,
+        )
+        featured_doc = await self.find_one_for_date(local_date)
+        return await self.featured(featured_doc, query_string, from_offset)
+
+    async def featured(
+        self, featured_doc: Optional[dict], query_string: Optional[str] = None, from_offset: int = 0
+    ) -> Any:
+        """Return featured items.
+
+        :param Optional[dict] featured_doc: The featured document for the given date
+        :param Optional[str] query_string: Optional search query to filter the results
+        :param int from_offset: Pagination offset for the results
+        :return: A list of filtered featured items
+        """
+        from newsroom.section_filters.service import SectionFiltersService
+
+        user = get_user_from_request(None)
+        company = get_company_from_request(None)
+        if is_events_only_access(user.to_dict(), company.to_dict()):  # type: ignore
+            abort(403)
+
+        if not featured_doc or not featured_doc.get("items"):
+            return []
+
+        query = build_agenda_query()
+        await SectionFiltersService().apply_section_filter(query, self.section)
+
+        planning_items_query = nested_query(
+            "planning_items",
+            {"bool": {"filter": [{"terms": {"planning_items.guid": featured_doc["items"]}}]}},
+            name="featured",
+        )
+
+        if query_string:
+            query["bool"]["filter"].append(self.query_string(query_string))
+            planning_items_query["nested"]["query"]["bool"]["filter"].append(planning_items_query_string(query_string))
+
+        query["bool"]["filter"].append(planning_items_query)
+
+        source = {"query": query, "size": len(featured_doc["items"]), "from": from_offset}
+        # self.set_post_filter(source, req)  # TODO: Confirm usage of function with previous ParsedReq
+        if not from_offset:
+            source["aggs"] = aggregations
+
+        if company and not is_admin(user) and company.get("events_only", False):
```

Comment:
`company` here is an instance of `CompanyResource`, and therefor you can just use `company.events_only`